setup.py

```diff
@@ -1,18 +1,30 @@
 #!/usr/bin/env python
+import json
 import os
 import sys
 
+from packaging.version import parse
 from setuptools import find_packages, setup
 
-min_version, max_version = ((3, 9), "3.9"), ((3, 11), "3.11")
-if not (min_version[0] <= sys.version_info[:2] <= max_version[0]):
-    # Python 3.5 does not support f-strings
+f = open("python_versions.json")
+supported_python_versions = json.load(f)
+f.close()
```

Comment:
I assume we're not using a context manager for maximal backwards compatibility?

Reply:
Yeah, though support for that probably goes way back? idk, was just kinda afraid to go that route